cmds/exp/sed/sed.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2023 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// By Ahmed Kamal <email.ahmedkamal@googlemail.com>
+
+// sed edits file contents using regular expressions.
+//
+// Synopsis:
+//
+//	sed [-ie] [FILE]...
+//
+// Options:
+//
+//	-i, --in-place        in-place file edit
+//	-e, --expression      edits to perform (only s/foo/bar style supported)
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+
+	flag "github.com/spf13/pflag"
+)
+
+// Writes to a temp file first
+// Renames to the final file upon closing
+type tmpWriter struct {
+	ftmp     os.File
+	filename string
+}
+
+func newTmpWriter(filename string) (*tmpWriter, error) {
+	ftmp, err := os.CreateTemp("/tmp", ".sed*.txt")
+	if err != nil {
+		return nil, fmt.Errorf("unable to create temp file: %w", err)
+	}
+	return &tmpWriter{filename: filename, ftmp: *ftmp}, nil
+}
+
+func (tw *tmpWriter) Write(b []byte) (int, error) {
+	return tw.ftmp.Write(b)
+}
+
+func (tw *tmpWriter) Close() error {
+	err := tw.ftmp.Close()
+	if err != nil {
+		return err
+	}
+	os.Rename(tw.ftmp.Name(), tw.filename)
+	return nil
+}
+
+type transform struct {
+	from   *regexp.Regexp
+	to     string
+	global bool
+}
+
+func (t *transform) run(input io.Reader) io.Reader {
+	pr, pw := io.Pipe()
+	scanner := bufio.NewScanner(input)
+	go func() {
+		defer pw.Close()
+		for scanner.Scan() {
+			line := scanner.Text()
+			var replaced string
+			if t.global {
+				replaced = t.from.ReplaceAllString(line, t.to)
+			} else {
+				counter := 0
+				// Replaces only the first occurrence which golang stdlib does not offer!
```

Comment:
I think you can use both regex and strings to simplify:
found := t.from.FindString(line)
if found != "" {
    strings.Replace(line, f, t.to, 1)
}